### ISSUE TITLE
Weird behavior in Schedule Detail block - Schedule Builder - related to time-change Sundays

### ISSUE THREAD

**Issue by stanalyst** [SparkDevNetwork/Rock#5801]
### Description

_Mark this one down in the obscure bug file..._
The new obsidian version of the schedule detail/schedule builder has a bug which impacts the start date/time input fields ... but this only occurs when the start date is set to a time-change Sunday. For most start dates entered - the time field works as expected. But when the start date is set to a time-change Sunday (like 11/5/2023 or 3/10/2024) - then the behavior of the date and time fields is NOT as expected ... often automatically changing the values in those fields or limiting what can be entered.

### Actual Behavior

If the start time is set to a fall time-change Sunday (like 11/5/2023) ... then the time block will not accept any time greater than 1:59am. If you enter a time of 2:00 or later ... then when you tab out of that input field - it will automatically change the hour to 1 am but keep the minute portion.

If the start time is set to a spring time-change Sunday (like 3/10/2024) ... and you enter a time of 2:00am or later - then the date will automatically be changed to the next day (3/11 in our example) ... and the time will change to 12:00 am.

### Expected Behavior

The schedule builder should not be adjusting the date/time fields at all. Let the user enter the start date/time they want. In other words - should be able to set the start date/time to 11/3/2023 10:00 am ... without the block changing it to 11/3/2023 1:00 am.

### Steps to Reproduce

- Edit a schedule from the General Settings/Schedules page
- Click Edit Schedule to enter the schedule builder
- Enter 11/5/2023 for the start date ... try setting the time to 7:00 am
- Observe behavior of block ... note that time changed to 1:00 am.

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

1.16.2.1

### Client Culture Setting

en-US

**Comment by chead4** [comment]
@stanalyst Hi Stan I am unable to recreate this issue on both the demo site (v16.4) and on v16.2.  Could you confirm if you're able to recreate this issue on the demo site and if so provide a video demonstrating the issue?

**Comment by stanalyst** [comment]
Sure ... here's a video showing the behavior I described ... from the demo site ... https://app.screencast.com/dUfl8Aeospnht

**Comment by chead4** [comment]
@stanalyst Hi Stan Thanks for taking the time to create the video; unfortunately, I am still unable to recreate the issue.  Can you provide details on your OS and Browser make/version?

**Comment by stanalyst** [comment]
It took me a minute - but I think I know why I'm seeing the behavior but you aren't. My guess is you are based in Arizona or somewhere that doesn't observe daylight savings time. And the computer you are using to try and reproduce the issue is probably set to have a time zone of Mountain Standard Time. 

To confirm this - I changed my laptop (MacBook Pro) with the closest city being Phoenix which set the time zone to Mountain Standard time - and then tried to enter a schedule with a start date of 11/5/2023 (a time-change Sunday) and start time of 8:00am  - and the behavior did NOT occur. 

Then I changed my closest city to Denver - which set the time zone to Mountain Daylight time - and tried to enter a schedule with the same settings - and the behavior occurred ... automatically changing the time to 1:00am.
